org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/indexing/SourceIndexer.java

```diff
@@ -88,6 +98,10 @@ public SourceIndexer(SearchDocument document) {
 	}
 	@Override
 	public void indexDocument() {
+		if (Boolean.getBoolean(getClass().getSimpleName() + ".DOM_BASED_INDEXER")) { //$NON-NLS-1$
```

Comment:
Why is this flag per class vs a project or workspace preference?  For a large project, this could result in a lot of system properties.  As well, why are you are using the simple name as there could be collision with similarly named classes in different packages.

Reply:
Using a simple name is for consistency with other system properties such as `ICompilationUnitResolver` which are already used in cases. I can switch it to fully qualified name if this is a must-have.
Note that we do not want to offer it as a preference or something more usable. This switch is deep in the code so that it's mostly known of people who maintain the code, and will not be part of any API supported by JDT. Sometimes in the future, the switch will be removed together with legacy code.

Reply:
Ok